src/incubator/WheelPicker/index.tsx

```diff
@@ -2,90 +2,164 @@
 
 export interface WheelPickerProps {
+  /**
+   * Data source for WheelPicker
+   */
   items?: ItemProps[];
+  /**
+   * Describe the height of each item in the WheelPicker
+   */
   itemHeight?: number;
-  itemTextStyle?: StyleProp<TextStyle>;
-  onChange: (item: ItemProps, index: number) => void;
+  /**
+   * TextStyle for the focused row
+   */
+  activeItemTextStyle?: StyleProp<TextStyle>;
+  /**
+   * TextStyle for other, non-focused rows
+   */
+  inactiveItemTextStyle?: StyleProp<TextStyle>;
+  /**
+   * Event, on active row change
+   */
+  onChange: (index: number, item?: ItemProps) => void;
 }
 
-const WheelPicker = ({items, itemHeight = 48, itemTextStyle}: WheelPickerProps) => {
-  const height = itemHeight * 4;
-  const scrollview = useRef<Animated.ScrollView>();
+type WrappedItem = {
+  isActive: boolean;
+} & ItemProps;
+
+const WheelPicker = ({items, itemHeight = 48, activeItemTextStyle, inactiveItemTextStyle, onChange: onChangeEvent}: WheelPickerProps) => {
+  const height = itemHeight * 5;
+  const scrollView = useRef<Animated.ScrollView>();
   const [offset] = useValues([0], []);
-  const onScroll = onScrollEvent({y: offset});
+  const activeIndex = useRef<number>(0);
+
+  const wrapItems = (items?: ItemProps[]): WrappedItem[] => {
+    return _.map(items, (item, index: number) => {
+      return {...item, isActive: activeIndex.current === index};
+    });
+  };
+
+  const [itemsWrap, setItemWrapped] = useState<WrappedItem[]>(wrapItems(items));
+
+  const updateItems = () => {
+    setItemWrapped(wrapItems(items));
+  };
 
   const selectItem = useCallback(
     index => {
-      if (scrollview.current) {
-        scrollview.current.getNode().scrollTo({y: index * itemHeight, animated: true});
+      if (scrollView.current?.getNode()) {
+        //@ts-ignore for some reason scrollToOffset not recognized
+        scrollView.current.getNode().scrollToOffset({offset: index * itemHeight, animated: true});
       }
     },
     [itemHeight]
   );
 
   const onChange = useCallback(() => {
-    // TODO: need to implement on change event that calc the current selected index
-  }, [itemHeight]);
+    onChangeEvent(activeIndex.current, items?.[activeIndex.current]);
+  }, []);
+
+  const valueInRange = (value: number, min: number, max: number): number => {
+    if (value < min || value === -0) {
+      return min;
+    }
+    if (value > max) {
+      return max;
+    }
+    return value;
+  };
+
+  const calculateCurrentIndex = (offset: number): number => {
+    const calculatedIndex = Math.round(offset / itemHeight);
+    return valueInRange(calculatedIndex, 0, items!.length - 1);
+  };
+
+  const onScroll = useCallback((event: NativeSyntheticEvent<NativeScrollEvent>) => {
+    const currentIndex = calculateCurrentIndex(event.nativeEvent.contentOffset.y);
+    if (activeIndex.current != currentIndex) {
+      activeIndex.current = currentIndex;
+      updateItems();
+    }
```

Comment:
This will set all of the items whenever the index change. 
Which happens a lot in wheel pickers and in a very rapid pace. 
I think it can be very problematic for performance.